packages/lib-classifier/src/store/AnnotatedSteps/AnnotatedSteps.js

```diff
@@ -27,45 +41,82 @@ const AnnotatedSteps = types.model('AnnotatedSteps', {
 .actions(self => {
   setUndoManager(self)
 
+  /** Create a new history entry from the current active step. **/
+  function _createStep() {
+    const { classifications, workflowSteps } = getRoot(self)
+    const classification = tryReference(() => classifications?.active)
+    const step = tryReference(() => workflowSteps.active)
+    if (classification && step) {
+      const { history, steps } = self
+      let annotations
+      history.withoutUndo(() => {
+        annotations = step.tasks.map(task => classification.createAnnotation(task))
+      })
+      const historyStep = {
+        id: cuid(),
+        step,
+        annotations
+      }
+      console.log(`Adding ${historyStep.step.stepKey} to history`)
+      steps.put(historyStep)
+    }
+  }
+  /** Clear the redo history and restart history from this point with a new step. */
+  function _replace() {
+    const { history } = self
+    const { workflowSteps } = getRoot(self)
+    history.undo()
+    history.clearRedo()
```

Comment:
This leaves orphaned annotations in `classification.annotations`. It would be good if those could be cleaned up when we replace history.

Reply:
I guess the MST answer to this would be that `step.annotations` should store the annotations for a step, not references. Safe references will be automatically cleaned up when the original is destroyed.

Reply:
The downside of this might be that every letter in a text annotation gets recorded as a new undo step in the history stack.